dbt_coves/tasks/generate/sources.py

```diff
@@ -66,16 +70,70 @@ def register_parser(cls, sub_parsers, base_subparser):
             help="Folder with jinja templates that override default "
             "sources generation templates, i.e. 'templates'",
         )
+        subparser.add_argument(
+            "--metadata",
+            type=str,
+            help="Path to csv file containing metadata, i.e. 'metadata.csv'",
+        )
         subparser.set_defaults(cls=cls, which="sources")
         return subparser
 
+    def get_metadata(self, path):
+        """
+        If metadata path was provided, it returns a dictionary with column keys and their corresponding values
+        """
+        if path:
+            metadata_path = Path().joinpath(path)
+            try:
+                f = open(metadata_path, "r")
+            except OSError:
+                raise ValueError(
+                    f"Failed to load metadata. File {metadata_path} not found or corrupt. Please, try again."
+                )
+            with f as csvfile:
+                rows = csv.DictReader(csvfile)
+                metadata_map = dict()
+                for row in rows:
+                    mdata_key = (
+                        row["database"]
+                        + "-"
+                        + row["schema"]
+                        + "-"
+                        + row["relation"]
+                        + "-"
+                        + row["column"]
+                        + "-"
+                        + row.get("key")
+                    )
+                    if not mdata_key in metadata_map:
+                        if mdata_key[-1] == "-":
+                            mdata_key = mdata_key[:-1]
+                            name = row["column"].lower()
+                            metadata_map[mdata_key] = {
```

Comment:
Looks like duplicated code with lines 120-123. And type is lowered here, but not there.